### PR TITLE
COM-19911: Implement the Blog full view

### PR DESCRIPTION
**Pull request by clash82**
https://jira.ez.no/browse/COM-19911

Desktop view:
fireshot capture 15 - blog - ez platform - http___ezplatform lh_blog

### CODE REVIEW

File: app/Resources/views/full/home.html.twig [hunk 1]
```diff
@@ -1,21 +1,28 @@
 {% extends 'pagelayout.html.twig' %}
 
+{% block main %}
+    <main class="white-background">
```
Comment (SylvainGuittard) [inline app/Resources/views/full/home.html.twig]: Very good idea. +1

File: app/Resources/views/pagelayout.html.twig [hunk 2]
```diff
@@ -8,12 +8,15 @@
         <link rel="shortcut icon" href="{{ asset('favicon.ico') }}" type="image/x-icon">
         <link rel="stylesheet" href="//maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
         <link rel="stylesheet" href="//maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css">
+        <link rel="stylesheet" href="//fonts.googleapis.com/css?family=Open+Sans:400,600">
+        <link rel="stylesheet" href="//fonts.googleapis.com/css?family=Open+Sans+Condensed:300|Open+Sans:400,700">
```
Comment (SylvainGuittard) [inline app/Resources/views/pagelayout.html.twig]: According to the layout, we need Open sans Condensed **Bold**
300 is not bold ;)

File: app/Resources/views/pagelayout_header.html.twig [hunk 3]
```diff
@@ -9,14 +9,15 @@
                     <span class="icon-bar"></span>
                 </button>
 
-                <a class="navbar-brand" href="//ezplatform.com">
+                <a class="navbar-brand" href="/">
```
Comment (SylvainGuittard) [inline app/Resources/views/pagelayout_header.html.twig]: Should not be in this PR ;)

File: app/Resources/views/parts/blog/author.html.twig [hunk 4]
```diff
@@ -0,0 +1,14 @@
+<div class="blog-post-line-author-image">
+    {{ ez_render_field(content, 'profile_picture', {
+        'parameters': {
+            'alias': 'blog_post_author',
+            'class': 'img-responsive'
```
Comment (SylvainGuittard) [inline app/Resources/views/parts/blog/author.html.twig]: Do you really need ```img-responsive```?

File: app/config/views.yml [hunk 5]
```diff
@@ -10,6 +10,9 @@ ezpublish:
                     home:
                         controller: ez_query:LocationQueryAction
                         params:
+                            omitted_location_ids:
```
Comment (SylvainGuittard) [inline app/config/views.yml]: Better than ids, we can define content type identifiers.
This should be in a different PR

File: web/assets/images/loading.svg [hunk 6]
```diff
@@ -0,0 +1,7 @@
+<svg version="1.1" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 130 130">
```
Comment (SylvainGuittard) [inline web/assets/images/loading.svg]: I think you can optimize this one ;)
https://jakearchibald.github.io/svgomg/

File: app/Resources/views/full/home.html.twig (hunk 1 above)
Comment (clash82) [inline app/Resources/views/full/home.html.twig]: Thank you :-)

File: app/Resources/views/pagelayout.html.twig (hunk 2 above)
Comment (clash82) [inline app/Resources/views/pagelayout.html.twig]: There is no Open Sans Condensed Bold font :-)

File: app/Resources/views/pagelayout_header.html.twig (hunk 3 above)
Comment (clash82) [inline app/Resources/views/pagelayout_header.html.twig]: Yes, I know but it helps to link to homepage on localhost...

File: app/Resources/views/parts/blog/author.html.twig (hunk 4 above)
Comment (clash82) [inline app/Resources/views/parts/blog/author.html.twig]: No, we can skip it.

File: web/assets/images/loading.svg (hunk 6 above)
Comment (clash82) [inline web/assets/images/loading.svg]: Very good point :-)

File: app/config/views.yml (hunk 5 above)
Comment (clash82) [inline app/config/views.yml]: Yes, but this will require more work. Let's leave it for now and improve it later.

**Comment by clash82** [comment]
ping @damianz5 @kamilmusial

File: app/Resources/views/full/blog.html.twig [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+    </div>
+
+    <div class="container">
+        {% include 'parts/content_list.html.twig' with {
+            'items': posts,
+            'viewType': 'line'
+        } %}
+
+        <div class="row">
+            <div class="col-md-3">
```
Comment (kamilmusial) [inline app/Resources/views/full/blog.html.twig]: Why not `col-offset-md-3` in following `div`?

File: app/Resources/views/full/blog.html.twig [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+        </div>
+    </div>
+
+    <script>
+        {% autoescape 'js' %}
+            var page = 1,
+                loadMoreButton = $('.blog-load-more').html();
+
+            function loadBlogPosts() {
+                page = page+1;
```
Comment (kamilmusial) [inline app/Resources/views/full/blog.html.twig]: :nitpick: I'd go `page += 1;` :nitpick: ;)

File: app/Resources/views/full/blog.html.twig [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+
+                $('.blog-load-more').html('<div class="blog-load-progress"></div>');
+
+                $.ajax({
+                    type: 'POST',
+                    url: '/ezplatform/blog/load/page/'+page,
```
Comment (kamilmusial) [inline app/Resources/views/full/blog.html.twig]: Missing space around `+` (it's present in other js files)

File: app/Resources/views/full/home.html.twig [hunk 10]
```diff
@@ -1,21 +1,28 @@
...
+
 {% block content %}
     {% if items.searchHits is not empty %}
         {% for item in items.searchHits %}
-            {% if not loop.first %}
-                {% set delimiter = 'delimiter' %}
-            {% else %}
-                {% set delimiter = '' %}
+            {% if item.valueObject.id not in omitted_location_ids %}
+                {% if not loop.first %}
```
Comment (kamilmusial) [inline app/Resources/views/full/home.html.twig]: That's one line per condition, I'd use ternary operator here

File: app/Resources/views/full/blog.html.twig [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+                page = page+1;
+
+                $('.blog-load-more').fadeOut('fast', function () {
+                    $(this).html('<div class="blog-load-progress"></div>').fadeIn();
+                });
+
+
+                $.ajax({
+                    type: 'POST',
+                    url: '/ezplatform/blog/load/page/'+page,
```
Comment (kamilmusial) [inline app/Resources/views/full/blog.html.twig]: Missing spaces around `+` (present in other js parts)

File: app/Resources/views/full/blog.html.twig [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+            function loadBlogPosts() {
+                page = page+1;
+
+                $('.blog-load-more').fadeOut('fast', function () {
+                    $(this).html('<div class="blog-load-progress"></div>').fadeIn();
+                });
+
+                $.ajax({
+                    type: 'POST',
+                    url: '/ezplatform/blog/load/page/'+page,
```
Comment (kamilmusial) [inline app/Resources/views/full/blog.html.twig]: Missing spaces around `+` (present in other js parts)

File: app/Resources/views/line/blog_post.html.twig [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+        {% endif %}
+
+        <h2>
+            <a href="{{ path(location) }}">
+                {{ ez_render_field(content, 'title') }}
+            </a>
+        </h2>
+
+        <div class="blog-post-line-intro">
+            {{ ez_render_field(content, 'intro') }}
```
Comment (kamilmusial) [inline app/Resources/views/line/blog_post.html.twig]: Can't into also be a link? Then you'd get rid of extra `<a href="{{ path(location) }}">`s and wrap whole post in one `<a>`

File: app/Resources/views/full/home.html.twig (hunk 1 above)
Comment (kamilmusial) [inline app/Resources/views/full/home.html.twig]: Simple, yet brilliant ;)

File: src/AppBundle/EventListener/PreContentViewListener.php [hunk 14]
```diff
@@ -0,0 +1,28 @@
...
+
+    /**
+     * @param array $options
+     */
+    public function __construct(array $options = [])
+    {
+        $this->options = $options;
+    }
+
+    public function onPreContentView(PreContentViewEvent $event)
```
Comment (kamilmusial) [inline src/AppBundle/EventListener/PreContentViewListener.php]: Missing doc

File: src/AppBundle/Controller/BlogController.php [hunk 15]
```diff
@@ -0,0 +1,78 @@
...
+
+    /**
+     * @param \Symfony\Bundle\TwigBundle\TwigEngine $templating
+     * @param \eZ\Publish\API\Repository\SearchService $searchService
+     * @param \AppBundle\QueryType\ChildrenQueryType $childrenQueryType
+     * @param int $blogLocationId
+     * @param int $blogPostsLimit
+     */
+    public function __construct(
+        TwigEngine $templating,
```
Comment (damianz5) [inline src/AppBundle/Controller/BlogController.php]: could we use `Symfony\Component\Templating\EngineInterface` interface intead of `TwigEngine` class?

File: src/AppBundle/Controller/BlogController.php [hunk 16]
```diff
@@ -0,0 +1,78 @@
...
+
+        $query = $this->childrenQueryType->getQuery([
+            'parent_location_id' => $this->blogLocationId,
+            'limit' => $this->blogPostsLimit,
+            'offset' => $offset,
+        ]);
+        $searchResults = $this->searchService->findLocations($query);
+
+        $renderedContent = $this->templating->render('parts/content_list.html.twig', [
+            'items' => $searchResults,
```
Comment (damianz5) [inline src/AppBundle/Controller/BlogController.php]: could we pass collection of Locations instead of valueObjects? it will be more reusable

File: src/AppBundle/Resources/config/services.yml [hunk 17]
```diff
@@ -1,4 +1,13 @@
 services:
+    ezplatform.pre_content_view_listener:
+        class: AppBundle\EventListener\PreContentViewListener
+        arguments:
+            - blogLocationId: '%blog.location_id%'
+        tags:
+            - name: kernel.event_listener
+              event: ezpublish.pre_content_view
+              method: onPreContentView
```
Comment (damianz5) [inline src/AppBundle/Resources/config/services.yml]: could we use the same styling convention as symfony? 
[see app.exception_listener yaml example](http://symfony.com/doc/current/event_dispatcher.html)

File: app/Resources/views/full/blog.html.twig (hunk 7 above)
Comment (clash82) [inline app/Resources/views/full/blog.html.twig]: Very good point Kamil, I forgot about it!

File: app/Resources/views/full/blog.html.twig (hunk 8 above)
Comment (clash82) [inline app/Resources/views/full/blog.html.twig]: Second +1

File: app/Resources/views/line/blog_post.html.twig (hunk 13 above)
Comment (clash82) [inline app/Resources/views/line/blog_post.html.twig]: Nope, only title and thumbnail should be linked but the idea above is good.

File: app/Resources/views/full/home.html.twig (hunk 10 above)
Comment (clash82) [inline app/Resources/views/full/home.html.twig]: Yes, but it was implemented by @SylvainGuittard 😋  for the home view...

File: src/AppBundle/Controller/BlogController.php (hunk 15 above)
Comment (clash82) [inline src/AppBundle/Controller/BlogController.php]: Agree, should be like that.

File: src/AppBundle/Controller/BlogController.php (hunk 16 above)
Comment (clash82) [inline src/AppBundle/Controller/BlogController.php]: No we can't, it's related to how `QueryTypes` works when injected for the specific view.

File: app/Resources/views/full/home.html.twig (hunk 10 above)
Comment (kamilmusial) [inline app/Resources/views/full/home.html.twig]: Don't want to mess with @SylvainGuittard 😛

File: src/AppBundle/Resources/config/services.yml (hunk 17 above)
Comment (clash82) [inline src/AppBundle/Resources/config/services.yml]: I would prefer to keep like that because git-diff looks better in case of any further changes.

File: app/Resources/views/full/home.html.twig (hunk 10 above)
Comment (clash82) [inline app/Resources/views/full/home.html.twig]: better don't mess with a boss
67552e5a41a024754d298413e33cda42

**Comment by clash82** [comment]
OK guys I need your final approval for this PR.
/cc @kamilmusial @damianz5 @SylvainGuittard 

btw, I added a fadeIn/fadeOut effect and small animation when new blog posts are shown. You'll like it :)

File: app/Resources/views/full/home.html.twig (hunk 10 above)
Comment (kamilmusial) [inline app/Resources/views/full/home.html.twig]: > You come into my pull request on the day my code is to be deployed and you ask me to refactor - for money.

File: app/Resources/views/pagelayout.html.twig (hunk 2 above)
Comment (SylvainGuittard) [inline app/Resources/views/pagelayout.html.twig]: Bold is available ;)
https://fonts.google.com/specimen/Open+Sans+Condensed?selection.family=Open+Sans+Condensed:300,700

Comment (clash82) [inline app/Resources/views/pagelayout.html.twig]: I'm using `600` but can change it to `700`

Comment (SylvainGuittard) [inline app/Resources/views/pagelayout.html.twig]: You don't have the choice

**Comment by clash82** [comment]
ping @SylvainGuittard

**Comment by SylvainGuittard** [comment]
Tested with Insight
+1
